openmdao/utils/indexer.py

```diff
@@ -544,14 +544,22 @@ def as_array(self, copy=False, flat=True):
         ndarray
             The index array.
         """
-        # use maxsize here since a shaped slice always has positive int start and stop
-        arr = np.arange(*self._slice.indices(sys.maxsize), dtype=int)
-        if flat:
-            return arr
-
-        if self._orig_shape is None:
```

Comment:
Wow, this was totally broken. The variable `_orig_shape` isn't defined anywhere.